src/libraries/System.Net.Http/tests/FunctionalTests/ResponseStreamConformanceTests.cs

```diff
@@ -0,0 +1,133 @@
+            Assert.True(pair.Stream2.CanRead);
+            return pair;
+        }
+    }
+
+    public sealed class Http1ContentLengthResponseStreamConformanceTests : ResponseStandaloneStreamConformanceTests
+    {
+        protected override async Task WriteResponseAsync(Stream responseStream, byte[] bodyData)
+        {
+            await responseStream.WriteAsync(Encoding.ASCII.GetBytes($"HTTP/1.1 200 OK\r\nContent-Length: {bodyData.Length}\r\n\r\n"));
+            await responseStream.WriteAsync(bodyData);
+        }
+    }
+
+    public sealed class Http1ChunkedResponseStreamConformanceTests : ResponseStandaloneStreamConformanceTests
+    {
+        protected override async Task WriteResponseAsync(Stream responseStream, byte[] bodyData)
+        {
+            await responseStream.WriteAsync(Encoding.ASCII.GetBytes($"HTTP/1.1 200 OK\r\nTransfer-Encoding: chunked\r\n\r\n{bodyData.Length:X}\r\n"));
+            await responseStream.WriteAsync(bodyData);
+            await responseStream.WriteAsync(Encoding.ASCII.GetBytes("\r\n"));
+            await responseStream.WriteAsync(Encoding.ASCII.GetBytes("0\r\n\r\n"));
+        }
+    }
+
+    public abstract class ResponseConnectedStreamConformanceTests : ConnectedStreamConformanceTests
+    {
+        protected override Type UnsupportedConcurrentExceptionType => null;
+        protected override bool UsableAfterCanceledReads => false;
+
+        protected abstract string GetResponseHeaders();
+
+        protected override async Task<StreamPair> CreateConnectedStreamsAsync()
+        {
+            (Stream httpConnection, Stream server) = ConnectedStreams.CreateBidirectional(4096, int.MaxValue);
+
+            using var hc = new HttpClient(new SocketsHttpHandler() { ConnectCallback = delegate { return ValueTask.FromResult(httpConnection); } });
+            Task<HttpResponseMessage> clientTask = hc.SendAsync(new HttpRequestMessage(HttpMethod.Get, $"http://doesntmatter:12345/"), HttpCompletionOption.ResponseHeadersRead);
+
+            await ReadHeadersAsync(server);
+
+            byte[] responseHeader = Encoding.ASCII.GetBytes(GetResponseHeaders());
+            await server.WriteAsync(responseHeader);
+
+            return (server, (await clientTask).Content.ReadAsStream());
+        }
+
+        internal static async Task ReadHeadersAsync(Stream server)
+        {
+            var buffer = new byte[256];
+            string text = "";
+            while (!text.EndsWith("\r\n\r\n", StringComparison.Ordinal))
+            {
+                int bytesRead = await server.ReadAsync(buffer);
+                Assert.InRange(bytesRead, 1, buffer.Length);
+                text += Encoding.ASCII.GetString(buffer.AsSpan(0, bytesRead));
+            }
+        }
+
+        public override Task Disposed_ThrowsObjectDisposedException() =>
+            // The HTTP response streams don't throw ObjectDisposedException upon disposal.
+            Task.CompletedTask;
```

Comment:
Is this something we should fix?

Reply:
We could.  In general I'm not personally particularly concerned with behavior of objects after they've been disposed.